### PR TITLE
Add new metadata fields and add x-api-key header to eth-bytecode-db r…

### LINKED ISSUE #8720: Add smart-contracts metadata to the eth-bytecode-db request
https://app.swaggerhub.com/apis/rimrakhimov/EthereumBytecodeDatabase/v2#/

### PR DESCRIPTION
**Pull request by nikitosing**
…equest

Close #8720 

## Changelog
- Support new metadata format
```
"metadata": {
    "blockNumber": "string",
    "chainId": "string",
    "contractAddress": "string",
    "creationCode": "string",
    "deployer": "string",
    "runtimeCode": "string",
    "transactionHash": "string",
    "transactionIndex": "string"
  }
```
- Add `x-api-key` header
- New env `MICROSERVICE_SC_VERIFIER_API_KEY` (API key for verification that metadata sent to verifier microservice from a trusted source)
https://github.com/blockscout/docs/pull/194
- Add `/api/v2/import/smart-contracts/{address_hash}` endpoint

## Checklist for your Pull Request (PR)

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

File: apps/block_scout_web/lib/block_scout_web/controllers/api/v2/import_controller.ex [hunk 1]
```diff
@@ -48,6 +52,33 @@ defmodule BlockScoutWeb.API.V2.ImportController do
     end
   end
 
+  def try_to_search_contract(conn, %{"address_hash_param" => address_hash_string}) do
```
Comment (vbaranov) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/v2/import_controller.ex:55]: Please add spec && doc

File: apps/block_scout_web/lib/block_scout_web/controllers/api/v2/fallback_controller.ex [hunk 2]
```diff
@@ -130,7 +130,7 @@ defmodule BlockScoutWeb.API.V2.FallbackController do
     |> render(:message, %{message: @restricted_access})
   end
 
-  def call(conn, {:already_verified, true}) do
+  def call(conn, {:already_verified, _}) do
```
Comment (vbaranov) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/v2/fallback_controller.ex:133]: What is the rationale of the change? Why do we still need to log that error, even If contract is not verified?

File: apps/explorer/lib/explorer/chain.ex [hunk 3]
```diff
@@ -3484,6 +3484,44 @@ defmodule Explorer.Chain do
     end
   end
 
+  def smart_contract_creation_tx_with_bytecode(address_hash) do
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain.ex:3487]: Please add spec && doc and move to separate module having in mind we're disbanding chain.ex

File: apps/explorer/lib/explorer/smart_contract/eth_bytecode_db_interface.ex [hunk 4]
```diff
@@ -17,6 +17,10 @@ defmodule Explorer.SmartContract.EthBytecodeDBInterface do
     end
   end
 
+  def search_contract_in_eth_bytecode_internal_db(%{"bytecode" => _, "bytecodeType" => _} = body) do
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/smart_contract/eth_bytecode_db_interface.ex:20]: Please add spec && doc

File: apps/explorer/lib/explorer/smart_contract/helper.ex [hunk 5]
```diff
@@ -135,4 +135,45 @@ defmodule Explorer.SmartContract.Helper do
         nil
     end
   end
+
+  def fetch_data_for_verification(address_hash) do
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/smart_contract/helper.ex:139]: Please add spec && doc

File: apps/block_scout_web/lib/block_scout_web/controllers/api/v2/fallback_controller.ex (hunk 2 above)
Comment (nikitosing) [reply apps/block_scout_web/lib/block_scout_web/controllers/api/v2/fallback_controller.ex:133]: This change is required because I added new with clause (which has `{:already_verified, something}` format) here: https://github.com/blockscout/blockscout/blob/np-support-metadata-for-eth-bytecode-db/apps/block_scout_web/lib/block_scout_web/controllers/api/v2/import_controller.ex#L61 
but here returned value won't be boolean, so to avoid duplicating of `call/2` clauses , I just grounded the second tuple item.

> Why do we still need to log that error, even If contract is not verified?

We need to cut logging from this module at all, as we discussed some time ago (https://blockscout.slack.com/archives/D0225V1SJ81/p1696877058531439)

**Review by vbaranov — APPROVED** [review]
